gson/src/test/java/com/google/gson/JsonParserTest.java

```diff
@@ -116,4 +118,18 @@ public void testReadWriteTwoObjects() throws Exception {
     BagOfPrimitives actualTwo = gson.fromJson(element2, BagOfPrimitives.class);
     assertEquals("two", actualTwo.stringValue);
   }
+
+  public void testPeekEmptyStream() throws IOException {
+    JsonReader reader = new JsonReader(new StringReader(""));
+    try {
+      assertEquals(JsonToken.END_DOCUMENT, reader.peek());
+      fail();
+    } catch (java.io.EOFException expected) { }
```

Comment:
Might be good to add an import for `java.io.EOFException` instead of using its qualified name here.